packages/react-components/react-positioning/src/types.ts

```diff
@@ -1,26 +1,37 @@
-import * as PopperJs from '@popperjs/core';
 import * as React from 'react';
 
+type Rect = {
+  width: number;
+  height: number;
+  x: number;
+  y: number;
+};
+
 export type OffsetFunctionParam = {
-  popper: PopperJs.Rect;
-  reference: PopperJs.Rect;
-  placement: PopperJs.Placement;
+  positionedRect: Rect;
+  targetRect: Rect;
+  position: Position;
+  alignment?: Alignment;
 };
 
-export type OffsetFunction = (param: OffsetFunctionParam) => [number | null | undefined, number | null | undefined];
+export type OffsetObject = { crossAxis?: number; mainAxis: number };
+
+export type OffsetShorthand = number;
 
-export type Offset = OffsetFunction | [number | null | undefined, number | null | undefined];
+export type OffsetFunction = (param: OffsetFunctionParam) => OffsetObject | OffsetShorthand;
+
+export type Offset = OffsetFunction | OffsetObject | OffsetShorthand;
 
 export type Position = 'above' | 'below' | 'before' | 'after';
 export type Alignment = 'top' | 'bottom' | 'start' | 'end' | 'center';
 
 export type AutoSize = 'height' | 'height-always' | 'width' | 'width-always' | 'always' | boolean;
 
-export type Boundary = PopperJs.Boundary | 'scrollParent' | 'window';
+export type Boundary = HTMLElement | Array<HTMLElement> | 'clippingParents' | 'scrollParent' | 'window';
 
-export type PopperRefHandle = {
+export type PositioningImperativeRef = {
```

Comment:
Nit: Consider PositioningRef.  Is the word imperative necessary?  Aren't all ref's imperative? :) Do you think there will be other kinds of positioning refs that we'll need to distinguish between?